packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -28,31 +32,57 @@ export const NotificationsContextProvider = ({
   children,
   unreadCount = 0,
 }: NotificationsContextProviderProps): ReactElement => {
+  const isExtension = !!process.env.TARGET_BROWSER;
   const { user } = useContext(AuthContext);
-  const [hasPermission, setHasPermission] = useState(
-    globalThis.window?.Notification?.permission === 'granted',
-  );
-
-  const notificationsAvailable = () => {
-    return !!globalThis.window?.Notification;
-  };
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [hasPermission, setHasPermission] = useState(false);
 
-  const requestPermission = async (): Promise<NotificationPermission> => {
+  const onTogglePermission = async (): Promise<NotificationPermission> => {
     if (!user) return 'default';
 
+    if (hasPermission) {
+      await OneSignalReact.setSubscription(false);
+      setHasPermission(false);
+      return 'denied';
+    }
+
     const result = await globalThis.window?.Notification?.requestPermission();
-    setHasPermission(result === 'granted');
+    const isGranted = result === 'granted';
+    setHasPermission(isGranted);
+    OneSignalReact.setSubscription(isGranted);
+
     return result;
   };
 
-  const data = useMemo(() => {
-    return {
+  useEffect(() => {
+    if (isInitialized || !user || isExtension) {
+      return;
+    }
+
+    OneSignalReact.init({
+      appId: process.env.NEXT_PUBLIC_ONESIGNAL_APP_ID,
+      allowLocalhostAsSecureOrigin: isDevelopment,
+      serviceWorkerParam: { scope: '/push/onesignal/' },
+    }).then(async () => {
+      setIsInitialized(true);
+      await OneSignalReact.setExternalUserId(user.id);
+      const isGranted = await OneSignalReact.getSubscription();
+      setHasPermission(isGranted);
+    });
+  }, [isInitialized, user]);
+
+  const data: NotificationsContextData = useMemo(
+    () => ({
+      isInitialized,
       unreadCount,
       hasPermission,
-      notificationsAvailable,
-      requestPermission,
-    };
-  }, [hasPermission, user]);
+      onTogglePermission,
+      get isNotificationAvailable() {
```

Comment:
I think "available" might be confusing with whether the user approved it or not

```suggestion
      get isNotificationSupported() {
```

Reply:
I agree. Let's do this.